### PR TITLE
Separate plots using #### in make_room_for_ylabel_using_axesgrid.py

### PR DESCRIPTION
**Pull request by ardieorden**
## PR Summary

Separate plots using ######### lines for Sphinx-gallery to render plots in sections.
Part of #8922.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py [hunk 1]
```diff
@@ -7,61 +7,65 @@
 from mpl_toolkits.axes_grid1 import make_axes_locatable
 from mpl_toolkits.axes_grid1.axes_divider import make_axes_area_auto_adjustable
 
+import matplotlib.pyplot as plt
 
-if __name__ == "__main__":
+def ex1():
+    plt.figure(1)
```
Comment (timhoffm) [inline examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py]:
```suggestion
    plt.figure()
```
We don't make any use of explicit figure numbers, so let's remove them to reduce clutter. Also in the other examples.

**Comment by ardieorden** [comment]
Thanks for the feedback and apologies for the late response! I removed the functions wrote them as top-level code and I also removed the figure numbers for each code block. Let me know if you'd like to see other changes!

File: examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py [hunk 2]
```diff
@@ -7,61 +7,56 @@
...
 from mpl_toolkits.axes_grid1.axes_divider import make_axes_area_auto_adjustable
 
+import matplotlib.pyplot as plt
 
-if __name__ == "__main__":
 
-    import matplotlib.pyplot as plt
+plt.figure()
+ax = plt.axes([0, 0, 1, 1])
+#ax = plt.subplot(111)
```
Comment (QuLogic) [inline examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py]: Might as well remove this.

File: examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py [hunk 3]
```diff
@@ -7,61 +7,56 @@
...
 
-        make_axes_area_auto_adjustable(ax1, pad=0.1, use_axes=[ax1, ax2])
-        make_axes_area_auto_adjustable(ax2, pad=0.1, use_axes=[ax1, ax2])
+make_axes_area_auto_adjustable(ax1, pad=0.1, use_axes=[ax1, ax2])
+make_axes_area_auto_adjustable(ax2, pad=0.1, use_axes=[ax1, ax2])
 
-    def ex3():
+###############################################################################
 
-        fig = plt.figure(3)
```
Comment (QuLogic) [inline examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py]: This `fig` is used later, but you've removed it.

**Comment by ardieorden** [comment]
Thanks for the code review @QuLogic ! I committed the changes for importing matplotlib and assigning the `fig` variable and I squashed it to the previous commit for cleanliness.

Let me know if you'd like to see other changes!

File: examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py (hunk 2 above)
Comment (timhoffm) [reply examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py]: Just remove the commented line.

**Comment by ardieorden** [comment]
Thanks for the comment @timhoffm ! I just added in the `import matplotlib.pyplot as plt` line above all the other imports and I removed the commented out line. Let me know if you'd like to request for more changes!

**Comment by timhoffm** [comment]
Thanks, and congratulations on your first contribution to Matplotlib! Hope to see you back some time.

**Pull request merged**